### PR TITLE
[MST-40]: JS SDK - Extend integration model type to include MS Teams type 👥 

### PR DESCRIPTION
**Pull request by sumitramanga**
### Description 📝 
This PR is to implement the MS Teams integration type and add an optional userId parameter to the delete integration endpoint

**Updates**
👉 Modify relevant enums to include MS Teams type
👉 Extend `IIntegration` to include the newly made `IMicrosoftTeamsIntegrationData` type
👉 Add new userId parameter to delete integration endpoint
👉 Bump package number

### Screenshots 📷
_Successfully setting up a new MS Teams integration and attempting to create another MS Teams integration on the same plan_
image

_Bad request (the parameters were out of order)_
image

### Test plan 🧪 
- Can pass through an MS Teams type into the integration endpoints
- `userId` parameter on delete integration endpoint is optional

### Author to check 👓 
- [ ] Builds pass
- [ ] Tested in an alternative environment
- [ ] Reviewed by another developer
- [ ] Appropriate documentation written (code comments, Quip docs)

### Reviewer to check ✔️ 
- [ ] Change has been tested in a separate environment
- [ ] Code is written to standards
- [ ] Appropriate tests have been written (code comments, Quip docs)

### CODE REVIEW

File: lib/entities/integrations/enums.ts [hunk 1]
```diff
@@ -2,5 +2,6 @@
 
 export enum IntegrationType {
   Webhook = "Webhook",
-  Slack = "Slack"
+  Slack = "Slack",
+  MsTeams = "MsTeams"
```
Comment (crystalqdm) [inline lib/entities/integrations/enums.ts]: I think we mightve made this decision when you were on break, but to standardize things we decided to use the full name `MicrosoftTeams` for any types/ class names etc. (:

File: lib/entities/integrations/index.ts [hunk 2]
```diff
@@ -70,6 +70,7 @@ export class Integrations {
    * Delete an integration matching the given identifier for the given plan
    * @param planIdentifier - Identifier of the target plan
    * @param identifier - Id related to a specific integration
+   * @param userId - Id of the user logged in
```
Comment (crystalqdm) [inline lib/entities/integrations/index.ts]: Can remove this

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@raygun.io/sdk",
   "displayName": "RaygunSDK",
-  "version": "0.1.61",
+  "version": "0.1.63",
```
Comment (crystalqdm) [inline package.json:4]: should it be `0.1.61` -> `0.1.62` ?

File: lib/entities/integrations/models.ts [hunk 4]
```diff
@@ -12,11 +12,16 @@ export type ISlackIntegrationData = {
   botUserId: string |null;
 }
 
+export type IMicrosoftTeamsIntegrationData = {
+  teamId: string | null;
+  serviceUrl: string | null;
```
Comment (crystalqdm) [inline lib/entities/integrations/models.ts:17]: The reason `teamId` can be a null type is because it is a sensitive data property and may come back as null from the public API. `serviceUrl` is not a sensitive property so should never have the null type

**Review by crystalqdm — CHANGES_REQUESTED** [review]
Can you please also test the GET endpoint to verify the `teamId` property comes back as null from the public api (as it is a sensitive data property)

File: lib/entities/integrations/models.ts (hunk 4 above)
Comment (sumitramanga) [reply lib/entities/integrations/models.ts:17]: Ahhh, that makes sense. Thank you for clarifying that!

File: lib/entities/integrations/enums.ts (hunk 1 above)
Comment (sumitramanga) [reply lib/entities/integrations/enums.ts]: Ah yes, I think I was referring to an old PR. Thanks for that

**Review by crystalqdm — APPROVED** [review]
Thanks for the changes! LGTM 🚀

**Review by crystalqdm — APPROVED** [review]
🚀
